setup/src/main.ts

```diff
@@ -1,20 +1,44 @@
 import * as core from '@actions/core';
 import * as exec from '@actions/exec';
+import * as github from '@actions/github';
+import axios from 'axios';
 
+const SPRKL_RECIPES = ['auto', 'uncommitted', 'mine', 'recent', 'all', 'lastPush', 'commitsList'];
+
+if (require.main === module) {
+    main();
+}
 
 /**
-    Setup sprkl function
+    Setup sprkl function.
  */
-(async () => {
+async function main() {
     // get the input values from the action
     const sprklVersion = core.getInput('version');
     const analyze = core.getInput('analyze');
     const setEnv = core.getInput('setenv');
+    const recipe = core.getInput('recipe');
+    const eventName = github.context.eventName;
+
+    // validate the inputs from the action user(only analyze, setEnv and recipe. No vaildation for sprklVersion)
+    validateInputOrFail(analyze, setEnv, recipe);
 
     // run sprkl install command
     const installCmd = `npx @sprkl/scripts@${sprklVersion} install`;
     await exec.exec(installCmd);
 
+    if (recipe === 'auto') {
```

Comment:
I don't like string games, I would prefer an empty string or undefined.

Reply:
@erez-sprkl what do you think? we talked about the default 'auto' option

Reply:
seems okay to me ..